### ISSUE TITLE
Ugly object destructing and function

### ISSUE THREAD

**Issue by sourabhv** [prettier/prettier#5928]
I updated to latest version of prettier today and prettier added these type of changes

```diff
-    reducer(state, { payload: { id, aspects } }) {
+    reducer(
+      state,
+      {
+        payload: { id, aspects },
+      }
+    ) {
```

```diff
-export default connect(mapStateToProps, mapDispatchToProps)(Account);
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(Account);
```

Which looks ugly, makes it harder to read (first one especially) and IMHO is kinda unnecessary.

Can we add an option to disable this if it's within printWidth?

___

Prettier RC:

```
---
singleQuote: true
trailingComma: es5
```

**Comment by ackvf** [comment]
I am having an increasing amount of issues with prettier and I am thinking about removing it from our project and instead set a strong set of eslint/tslint rules...

**Comment by brainkim** [comment]
@evilebottnawi The first issue might be a duplicate of #5249. The second issue is solved by #6033

**Comment by duailibe** [comment]
The first is not related to #5249, but to object destructuring. We have a few issues open, but not specifically about destructuring in function arguments.

Please in the future do not mangle different topics in a same issue, that makes it harder for us to manage
